addons/grenadethrowing/functions/fnc_pfh.sqf

```diff
@@ -0,0 +1,167 @@
+    if (count _currentThrowable < 1) exitWith {
+        [_unit, "No valid throwables"] call FUNC(exitThrowMode);
+    };
+    if (_currentThrowable select 0 == "") exitWith {
+        [_unit, "No valid throwables (check 2)"] call FUNC(exitThrowMode);
+    };
+
+    if (cameraView != "INTERNAL") exitWith {
+        [_unit, "Went into sighted mode"] call FUNC(exitThrowMode);
+    };
+
+    //@todo: Check to see if we have a grenade like that in our inventory still
+
+    private _currentGrenade = getText (configFile >> "CfgMagazines" >> _currentThrowable select 0 >> "ammo");
+    private _currentGrenadeMagazineType = _currentThrowable select 0;
+    GVAR(CurrentThrowSpeed) = getNumber (configFile >> "CfgMagazines" >> _currentThrowable select 0 >> "initSpeed");
+
+    if (isNil QGVAR(CurrentThrowSpeed)) then {
+        GVAR(CurrentThrowSpeed) = 0
+    };
+
+    // These magazine checks are potentially not needed, but maaaaaybe would become relevant in some situations, like someone taking a grenade from your pack if you were holding it.
+    if (GVAR(LastGrenadeTypeChecked) == "") then {
+        GVAR(ActiveGrenadeType) = _currentGrenade;
+
+        if (_currentGrenadeMagazineType in (magazines _unit)) then {
+            GVAR(ActiveGrenadeItem) = _currentGrenade createVehicleLocal ((vehicle _unit) modelToWorldVisual [0, 0.3, 1.6]);
+            if (GVAR(ActiveGrenadeType) == "") exitWith {
+                GVAR(CancelThrow) = false;
+                GVAR(GrenadeInHand) = false;
+            };
+            GVAR(ActiveGrenadeItem) enableSimulation false;
+        } else {
+            _exit = true;
+        };
+    };
+
+    if (_exit) exitWith {
+        [_unit, "Trying to create a grenade we do not have in inventory (1)"] call FUNC(exitThrowMode);
+    };
+
+    if (GVAR(LastGrenadeTypeChecked) != _currentGrenade) then {
+        deleteVehicle GVAR(ActiveGrenadeItem);
+        GVAR(ActiveGrenadeType) = _currentGrenade;
+        if (_currentGrenadeMagazineType in (magazines _unit)) then {
+            GVAR(ActiveGrenadeItem) = _currentGrenade createVehicleLocal ((vehicle _unit) modelToWorldVisual [0, 0.3, 1.6]);
+            if (GVAR(ActiveGrenadeType) == "") exitWith {
+                GVAR(CancelThrow) = false;
+                GVAR(GrenadeInHand) = false
+            };
+            GVAR(CurrentThrowSpeed) = getNumber (configFile >> "CfgMagazines" >> _currentThrowable select 0 >> "initSpeed");
+            GVAR(ActiveGrenadeItem) enableSimulation false;
+        } else {
+            _exit = true
+        };
+    };
+
+    if (_exit) exitWith {
+        [_unit, "Trying to create a grenade we do not have in inventory (2)"] call FUNC(exitThrowMode);
+    };
+
+    GVAR(CameraOffset) = [0, 0, 0.3];
+    private _posLeftOffset = 0;
+    private _posRightOffset = 0;
+
+    if (inputAction "leanLeft" > 0.1) then {
```

Comment:
we could probably tilt the throw point based on the position of the head in relation to the hip position or something like than. That could probably be usefull for side stances too.